### PR TITLE
ch4/ofi: fix support FI_HMEM

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
Make `MPIDI_OFI_ENABLE_HMEM` always a runtime setting so users can switch it off even when provider is set at configure time.

Initialize the hints with FI_HMEM capability bits. Most providers will only support FI_HMEM when the hint it explicitly provided.

[skip warnings]
## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch4/ofi
test:mpich/ch4/gpu/ofi

**Comment by hzhou** [comment]
test:mpich/ch4/ofi
test:mpich/ch4/gpu/ofi

**Comment by hzhou** [comment]
test:mpich/ch4/ofi
test:mpich/ch4/gpu/ofi

**Comment by hzhou** [comment]
test:mpich/ch4/ofi
test:mpich/ch4/gpu/ofi

**Review by abrooks98 — APPROVED** [review]
LGTM

File: src/mpid/ch4/netmod/ofi/init_settings.c [hunk 1]
```diff
@@ -215,6 +215,14 @@ int MPIDI_OFI_init_hints(struct fi_info *hints)
         /*     PROTOCOL         |  CONTEXT  |        SOURCE         |       TAG          */
         MPIDI_OFI_PROTOCOL_MASK | 0 | MPIDI_OFI_SOURCE_MASK | 0 /* With source bits */ :
         MPIDI_OFI_PROTOCOL_MASK | 0 | 0 | MPIDI_OFI_TAG_MASK /* No source bits */ ;
+
+    /* should we check MPIR_CVAR_ENABLE_GPU? */
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/init_settings.c:219]: I think yes, we should check the CVAR before enabling.

Comment (hzhou) [reply src/mpid/ch4/netmod/ofi/init_settings.c]: Updated.

**Pull request merged**